resources/testdriver-actions.js

```diff
@@ -417,5 +466,46 @@
     },
   };
 
+  function WheelSource() {
```

Comment:
I'd like to see a test under `infrastructure` that uses the wheel source, as a sanity check.